### DISCUSSION TITLE
Raspberry Pi/Dual Channel Expansion Hat/TS Dash/Ford PCM

### DISCUSSION THREAD

**Discussion by trebies** [collin80/SavvyCAN#772] [General]
Greetings, new to Github, SavvyCAN, and all things considered relatively new to coding/python as well.

I am working to interface with the PCM for one of Ford's 7.3/6R140 and display pertinent data to an LCD gauge cluster.  I grew up working in an auto shop and now work as a design engineer, thus the mechanical side of this endeavor comes naturally to me.  That said I only had a couple classes involving coding in college but have a younger brother interested in getting involved in Software Dev who has a similar background.

I figure that despite much research there may be some things that I don't realize I do not know.  I only recently stumbled across SavvyCAN from a video by Dan Dulac on youtube.  Because of this, I am completely open to suggestions from those more experienced with CAN interfaces.

Currently, I have configured a raspi3 with a dual channel CAN hat and gotten it to communicate with itself both in the terminal and using a python script.

Hat link:
https://seengreat.com/wiki/82/dual-ch-can-hat#toc11

I am working out kinks to sniff CAN data using this setup with a test vehicle directly from the 6/14 pins on the OBD2 port.

Question 1: Is it possible to run SavvyCAN live directly on the raspberrypi or would it be more intuitive to run it on a PC with some sort of OBD2 adapter?
Question 2: Reading through the other discussions and through the SavvyCAN documentation it sounds like a standard ELM327 OBD2 adapter (for use with programs like FORScan) may not be sufficient.  Is this the case?
Question 3: If the ELM327 isn't capeable and the RasPi isn't either, what adapter should I use?  I see that Dan Dulac uses the CSS CL1000.

Bonus Question: Beyond the questions above, my ultimate goal is to interface the learned CAN signals through an LCD display using software such as TSdash (from tuner studio) is there anyone who has experience with this?  Are there similar or better Dash configuration softwares out there?

Sorry for the apparent length of this "question" relative to other's posts, hopefully I am posting in the correct format and location, and maybe somebody finds my project interesting!

**Comment by collin80** [comment]
Q1: Yes, you can compile SavvyCAN and run it straight on an RPi. Many people have done it. It's my understanding that it might not be the easiest thing to set up. I have not done it myself. It would be nice if someone had posted binaries but I'm unaware of such a thing. Otherwise, you could capture traffic with an RPi and save it in the default output format of candump. SavvyCAN can read CANDump files.

Q2: Absolutely. Standard ELM327 devices are for very limited actions. They will not capture the entire bus load of an automobile. They are meant to send and receive to specific addresses only and ignore the rest. For reading the whole bus it'd be much better to have better hardware. An RPi with dual CAN hat counts as better hardware

Q3: The CSS CL1000 will work or PeakCAN or ValueCAN from Intrepid or kvaser Leaf Light. Or, if money is not burning a hole in your pocket, you can try cheap chinese ESP32 boards with CAN hardware. Chances are most of them could be made to work with ESP32RET. My actual employer makes hardware you can use: www.evtv.me the obvious choice would be an ESP32 board with dual CAN and CAN-FD. This is obviously the option that will offer the path of least resistance in using it with SavvyCAN but there are many, many devices that will work.
